### ISSUE TITLE
`redirectOn: 'no prefix'` does not redirect on the root URL

### ISSUE THREAD

**Issue by phanect** [nuxt-modules/i18n#2132]
### Environment

(Here's Stackblitz' environment information)

- Operating System: Linux
- Node Version:     v16.14.2
- Nuxt Version:     3.5.2
- Nitro Version:    2.4.1
- Package Manager:  npm@7.17.0
- Builder:          vite
- User Config:      ssr, components, modules, i18n
- Runtime Modules:  @nuxtjs/i18n@8.0.0-beta.12
- Build Modules:    -

### Reproduction

https://stackblitz.com/edit/nuxt-starter-2yxmzd

### Describe the bug

I added `strategy: 'prefix'` and `redirectOn: 'no prefix'` to nuxt.config.ts.

nuxt.config.ts

```ts
export default defineNuxtConfig({
  // ...
  i18n: {
    // ...
    strategy: 'prefix',
    // ...
    detectBrowserLanguage: {
      useCookie: true,
      cookieSecure: true,
      alwaysRedirect: true,
      fallbackLocale: 'en',
      redirectOn: 'no prefix',
    },
  },
});
```

If I understand correctly, with this config, if you open / (root URL), it should return 302 and redirect to /en or /ja. (Depending on your `Accept-Language`)
However, when you open / (root URL), it returns 404.

**Comment by phanect** [comment]
This issue might be related to #2131.

**Comment by phanect** [comment]
I guess the following code causes this issue:

https://github.com/nuxt-modules/i18n/blob/8dc25cbd88212563b5f9709740246a4915b8462f/src/runtime/utils.ts#L294-L299

`isI18nRouteDefined(route)` always returns `false` because `route.matched[0]?.meta.nuxtI18n` is always `undefined` on my end. (see the following `isI18nRouteDefined()` function:

https://github.com/nuxt-modules/i18n/blob/8dc25cbd88212563b5f9709740246a4915b8462f/src/runtime/utils.ts#L514-L517

Therefore, `detectRedirect()` returns `redirectPath` (=== empty string (`''`)) and [the following redirection process](https://github.com/nuxt-modules/i18n/blob/8dc25cbd88212563b5f9709740246a4915b8462f/src/runtime/utils.ts#L301-L348) is not executed.

I'm stuck here because I'm unsure how to make `route.matched[0]?.meta.nuxtI18n` a non-undefined value...

**Comment by kazupon** [comment]
Thanks!

You can workaround with `dedefaultLocale` option.

BTW, docs recommend that it is specified default locale
https://v8.i18n.nuxtjs.org/guide/routing-strategies#configuration

**Comment by phanect** [comment]
@kazupon Thanks for the information.
However, setting `defaultLocale` does not solve the problem.

In the following repro, I set `defaultLocale: "en"`.

https://stackblitz.com/edit/nuxt-starter-aj1bpj

However, `/` returns 200 and shows the English content, while it should return 302 to redirect to `/en` or `/ja` (Depending on your `Accept-Language`).

**Comment by urbgimtam** [comment]
@phanect Suffering the same issue.

**Comment by kazupon** [comment]
I seem that this issue is degraded by https://github.com/nuxt-modules/i18n/pull/1902
I'll try to revert that PR.

**Comment by phanect** [comment]
@kazupon Thanks so much for working on this! I confirmed this issue is partially fixed in v8.0.0-beta.13.

Now if you open https://example.com/, it automatically redirects to https://example.com/en.
However, it does not redirect to https://example.com/ja when you open the root page with `Accept-Language:
ja`. It always redirects to https://example.com/en.

Here's the repro.
https://github.com/phanect/repro-nuxt-i18n-redirection

Note: This sample code is available on [Stackblitz](https://stackblitz.com/edit/nuxt-starter-9w7cb2) too, but Stackblitz seems to have a bug as of now and it didn't work as expected. Please clone the repro and run it on your local machine.

Here's step to reproduce:

1. Run `npm run dev` at the project root of the repro.
2. Open your browser settings to set your primary language "Japanese" so that the `Accept-Language` header would be `ja`. This issue happens when `Accept-Language` is something like `ja,en;q=0.9,en-US;q=0.8` too.
3. Open any page under http://localhost:3000/, open the browser developer tools (Chrome: Application panel, Firefox: Storage panel), and remove `i18n_redirected` cookie.
4. Open http://localhost:3000/ (root page)

Expected: The browser redirects you to http://localhost:3000/ja
Actual: The browser redirects you to http://localhost:3000/en

Let me know if you prefer to close this issue and open another issue for `Accept-Language` bug.
